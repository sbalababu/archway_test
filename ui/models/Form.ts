export interface QuotaForm {
  quota: number;
}

export interface CoreMemoryForm {
  core: number;
  memory: number;
}

export interface MemberForm {
  username: string;
  roles: object;
}
