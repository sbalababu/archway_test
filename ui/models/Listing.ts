export interface Filters {
  filter: string;
  behaviors: string[];
  statuses: string[];
}
