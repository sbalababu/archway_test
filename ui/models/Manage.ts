import { ComplianceType } from '../constants';

export interface Question {
  question: string;
  requester: string;
  updated: Date;
  id?: number;
  complianceGroupId?: number;
}

export interface ComplianceContent {
  name: ComplianceType;
  id?: number;
  description: string;
  questions: Question[];
}

export interface Link {
  name: string;
  description: string;
  url: string;
  id?: number;
  customLinkGroupId?: number;
}

export interface LinksGroup {
  name: string;
  description: string;
  links: Link[];
  id?: number;
}
