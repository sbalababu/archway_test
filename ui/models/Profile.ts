export interface Profile {
  name: string;
  username: string;
  distinguished_name: string;
  permissions: {
    risk_management: boolean;
    platform_operations: boolean;
  };
}
