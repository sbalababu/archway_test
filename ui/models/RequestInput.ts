import { Compliance } from './Workspace';

export interface RequestInput {
  name: string;
  summary: string;
  description: string;
  compliance: Compliance;
}
