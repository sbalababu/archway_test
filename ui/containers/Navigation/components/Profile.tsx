import { Avatar } from 'antd';
import * as React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { createStructuredSelector } from 'reselect';
import { requestLogout } from '../../Login/actions';
import { getProfile } from '../../../redux/selectors';
import { getAuthType } from '../../Login/selectors';
import { Profile } from '../../../models/Profile';
import { SPNEGO } from '../../../constants';

interface Props {
  profile: Profile;
  authType: string;
  doLogout: () => void;
}

const ProfileComponent = ({ profile, authType, doLogout }: Props) => {
  const isSpnego: boolean = authType === SPNEGO;
  let name;
  if (profile && profile.name) {
    name = profile.name;
  }

  return (
    <div
      style={{
        width: '100%',
        padding: 15,
        color: 'white',
        textAlign: 'center',
        backgroundColor: '#415161',
        position: 'absolute' as 'absolute', // https://github.com/Microsoft/TypeScript/issues/11465
        bottom: 0,
      }}
    >
      <Avatar shape="circle" size="large" icon="user" style={{ backgroundColor: 'transparent', color: 'white' }} />
      <h3 style={{ color: 'white', textTransform: 'uppercase' }}>{name}</h3>
      {isSpnego || (
        <h6>
          <a href="#" style={{ textTransform: 'uppercase', fontWeight: 200, color: 'white' }} onClick={doLogout}>
            {' '}
            {/* eslint-disable-line */}
            log out
          </a>
        </h6>
      )}
    </div>
  );
};

const mapStateToProps = () =>
  createStructuredSelector({
    profile: getProfile(),
    authType: getAuthType(),
  });

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  doLogout: () => dispatch(requestLogout()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProfileComponent);
