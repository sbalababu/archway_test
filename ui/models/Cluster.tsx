import { Dropdown, Icon, Menu } from 'antd';
import * as Color from 'color';
import * as React from 'react';
import { Colors } from '../components';

export interface WebLocation {
  host: string;
  port: number;
}

export interface Statusable {
  status: string;
}

export class Status<T extends Statusable> {
  public statusable: T;

  constructor(statusable: T) {
    this.statusable = statusable;
  }

  public statusColor = (): Color => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return Colors.Green;
      case 'CONCERNING_HEALTH':
        return Colors.Orange;
      case 'BAD_HEALTH':
        return Colors.Red;
      default:
        return Colors.Gray;
    }
  };

  public statusText = () => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return 'good';
      case 'CONCERNING_HEALTH':
        return 'concerning';
      case 'BAD_HEALTH':
        return 'bad';
      default:
        return 'unknown';
    }
  };

  public glowColorText = () =>
    `0 0 5px 2px ${this.statusColor()
      .hsl()
      .string()}`;
}

export interface HueService extends Statusable {
  load_balancer: WebLocation[];
}

/* tslint:disable:no-empty-interface */
export interface HiveService extends Statusable {
  thrift: WebLocation[];
}

export interface YarnService extends Statusable {
  resource_manager: WebLocation[];
  node_manager: WebLocation[];
}

export interface NavigatorService extends Statusable {
  navigator: WebLocation[];
}

export abstract class ServiceLinks<T extends Statusable> {
  public service: T;

  public links!: JSX.Element[];

  constructor(service: T) {
    this.service = service;
  }
}

export class HueServiceLinks extends ServiceLinks<HueService> {
  public hueServiceLinks: JSX.Element[] =
    this.service &&
    this.service.load_balancer.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Hue UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.hueServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class HiveServiceLinks extends ServiceLinks<HiveService> {
  public links: JSX.Element[] = [0].map(index => (
    <h4 key={index} style={{ cursor: 'not-allowed', color: 'rgb(0,0,0,.45)', margin: 0 }}>
      (no links)
    </h4>
  ));
}

export class YarnServiceLinks extends ServiceLinks<YarnService> {
  public resourceManagerLinks: JSX.Element[] =
    this.service &&
    this.service.resource_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Resource Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public nodeManagerLinks: JSX.Element[] =
    this.service &&
    this.service.node_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Node Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown
      overlay={
        <Menu>
          {this.resourceManagerLinks}
          {this.nodeManagerLinks}
        </Menu>
      }
    >
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class NavigatorServiceLinks extends ServiceLinks<NavigatorService> {
  public navigatorServiceLinks: JSX.Element[] =
    this.service &&
    this.service.navigator.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Navigator UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.navigatorServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export interface Cluster extends Statusable {
  name: string;
  cm_url: string;
  services: {
    hive: HiveService;
    hue: HueService;
    yarn: YarnService;
    mgmt: NavigatorService;
  };
}
