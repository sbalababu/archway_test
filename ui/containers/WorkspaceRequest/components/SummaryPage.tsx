import * as React from 'react';
import { Card, Col, Icon, List, Row } from 'antd';
import Label from './Label';
import { ResourcePool, Workspace, Application, HiveAllocation } from '../../../models/Workspace';
import { Profile } from '../../../models/Profile';

interface Props {
  workspace?: Workspace;
  profile: Profile;
}

const ResourceItem = (name: string) => <h4>{name}</h4>;

const SummaryPage = ({ workspace, profile }: Props) => {
  const databaseNames = workspace && workspace.data.map((database: HiveAllocation) => database.name);
  const poolNames = workspace && workspace.processing.map((pool: ResourcePool) => pool.pool_name);
  const applicationNames = workspace && workspace.applications.map((application: Application) => application.name);
  return (
    <div>
      <h3>Here's a preview of what we'll request. Take a look and submit your request when you're ready!</h3>
      <Row type="flex" justify="center" style={{ marginTop: 25, marginBottom: 25 }}>
        <Col span={8}>
          <Card>
            <Icon type="crown" style={{ fontSize: 42 }} />
            <Label>You will be the owner for this project</Label>
            <h4>{profile.name}</h4>
          </Card>
        </Col>
      </Row>
      <Row type="flex" justify="center" gutter={25} style={{ marginTop: 25, marginBottom: 25 }}>
        <Col span={6} style={{ display: 'flex', flex: 1 }}>
          <Card style={{ justifyContent: 'center', display: 'flex', flex: 1 }}>
            <Icon type="database" style={{ fontSize: 42 }} />
            <Label>The following hive databases will be created for you</Label>
            <List renderItem={ResourceItem} dataSource={databaseNames} />
          </Card>
        </Col>
        <Col span={6} style={{ display: 'flex', flex: 1 }}>
          <Card style={{ justifyContent: 'center', display: 'flex', flex: 1 }}>
            <Icon type="rocket" style={{ fontSize: 42 }} />
            <Label>The following resource pools will be created for you</Label>
            <List renderItem={ResourceItem} dataSource={poolNames} />
          </Card>
        </Col>
        <Col span={6} style={{ display: 'flex', flex: 1 }}>
          <Card style={{ justifyContent: 'center', display: 'flex', flex: 1 }}>
            <Icon type="thunderbolt" style={{ fontSize: 42 }} />
            <Label>The following applications (with their own consumer group) will be created for you</Label>
            <List renderItem={ResourceItem} dataSource={applicationNames} />
          </Card>
        </Col>
      </Row>
    </div>
  );
};

export default SummaryPage;
