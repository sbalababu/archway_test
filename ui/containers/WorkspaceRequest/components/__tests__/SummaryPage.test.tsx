import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SummaryPage from '../SummaryPage';
import { Profile } from '../../../../models/Profile';

describe('SummaryPage', () => {
  it('renders correctly', () => {
    const profile: Profile = {
      name: '',
      username: '',
      distinguished_name: '',
      permissions: {
        risk_management: false,
        platform_operations: false,
      },
    };
    const wrapper = shallow(<SummaryPage profile={profile} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
