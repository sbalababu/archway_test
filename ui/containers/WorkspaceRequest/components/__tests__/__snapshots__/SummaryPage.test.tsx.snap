// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryPage renders correctly 1`] = `
<div>
  <h3>
    Here's a preview of what we'll request. Take a look and submit your request when you're ready!
  </h3>
  <Row
    gutter={0}
    justify="center"
    style={
      Object {
        "marginBottom": 25,
        "marginTop": 25,
      }
    }
    type="flex"
  >
    <Col
      span={8}
    >
      <Card>
        <Icon
          style={
            Object {
              "fontSize": 42,
            }
          }
          type="crown"
        />
        <Label>
          You will be the owner for this project
        </Label>
        <h4 />
      </Card>
    </Col>
  </Row>
  <Row
    gutter={25}
    justify="center"
    style={
      Object {
        "marginBottom": 25,
        "marginTop": 25,
      }
    }
    type="flex"
  >
    <Col
      span={6}
      style={
        Object {
          "display": "flex",
          "flex": 1,
        }
      }
    >
      <Card
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Icon
          style={
            Object {
              "fontSize": 42,
            }
          }
          type="database"
        />
        <Label>
          The following hive databases will be created for you
        </Label>
        <List
          bordered={false}
          dataSource={Array []}
          loading={false}
          pagination={false}
          renderItem={[Function]}
          split={true}
        />
      </Card>
    </Col>
    <Col
      span={6}
      style={
        Object {
          "display": "flex",
          "flex": 1,
        }
      }
    >
      <Card
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Icon
          style={
            Object {
              "fontSize": 42,
            }
          }
          type="rocket"
        />
        <Label>
          The following resource pools will be created for you
        </Label>
        <List
          bordered={false}
          dataSource={Array []}
          loading={false}
          pagination={false}
          renderItem={[Function]}
          split={true}
        />
      </Card>
    </Col>
    <Col
      span={6}
      style={
        Object {
          "display": "flex",
          "flex": 1,
        }
      }
    >
      <Card
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Icon
          style={
            Object {
              "fontSize": 42,
            }
          }
          type="thunderbolt"
        />
        <Label>
          The following applications (with their own consumer group) will be created for you
        </Label>
        <List
          bordered={false}
          dataSource={Array []}
          loading={false}
          pagination={false}
          renderItem={[Function]}
          split={true}
        />
      </Card>
    </Col>
  </Row>
</div>
`;
